### DISCUSSION TITLE
Partial File Transcription

### DISCUSSION THREAD

**Discussion by colvinem** [deepgram/community#257] [General help] [answered]
I'm trying to work on transcriptions of a podcast and I'm only using Deepgram for the parts that aren't scripted. My problem is that I'm not sure how to have Deepgram start at a specific timestamp rather than transcribing the entire thing (both I and the podcast are on pretty tight budgets, so we're trying not to do any more than we need to). My Python is functional but not the strongest (and also pretty specialized in other applications), but I've included my script below:

```
from deepgram import Deepgram
import asyncio
import json
import os
import sys

DEEPGRAM_API_KEY = ''

FILE = r'CHATWQT_mixdown.mp3'
MIMETYPE = 'audio/mpeg'

async def main():
	deepgram = Deepgram(DEEPGRAM_API_KEY)
	if FILE.startswith('http'):
		source = {
			'url': FILE
		}
	else:
		audio = open(FILE, 'rb')
		source = {
			'buffer': audio,
			'mimetype': MIMETYPE
		}
	response = await asyncio.create_task(
	  deepgram.transcription.prerecorded(
	    source,
	    {
	      'punctuate': True,
              'paragraphs': True,
	      'diarize': True,
	      'language': 'en-GB'
	    }
	  )
	)

	transcript = response['results']['channels'][0]['alternatives'][0]['paragraphs']['transcript']
	with open('QueerThings.txt','w') as f:
		f.write(transcript)
	f.close()
try:
	asyncio.run(main())
except Exception as e:
	exception_type, exception_object, exception_traceback = sys.exc_info()
	line_number = exception_traceback.tb_lineno
	print(f'line {line_number}: {exception_type} - {e}')
```

**Comment by jjmaldonis** [comment, marked as answer]
Hey @colvinem! You'll want to split your audio up into multiple files, based on the segments that you want Deepgram to transcribe. The most effective way to do this is by using `ffmpeg`, which has [python bindings](https://github.com/kkroening/ffmpeg-python).

The relevant command line command is `ffmpeg -ss START_TIME -t DURATION -i INPUT OUTPUT`, so that's what you'll want to emulate via the Python bindings if you use Python. You could also probably use the python `subprocess` module to call `ffmpeg` directly.
